STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/stm32l475_flash_driver.c

```diff
@@ -3,149 +3,129 @@
 // See LICENSE file in the project root for full license information.
 
 #include "stm32l475_flash_driver.h"
-#include "stm32l4xx.h"
 #include "stm32l475xx.h"
+#include "stm32l4xx.h"
 #include "stm32l4xx_hal.h"
 
 #include <stdint.h>
 
 /* Define the bank2 address for new firmware.  */
-#define FLASH_BANK2_ADDR                (FLASH_BASE + FLASH_BANK_SIZE)
+#define FLASH_BANK2_ADDR (FLASH_BASE + FLASH_BANK_SIZE)
 
 /* Define the internal variables.  */
-static uint32_t total_write_size;
-static uint32_t write_size;
-static union 
+static union
 {
-    uint64_t write_buffer_int64;
-    unsigned char write_buffer_char[8];
-}write_buffer;
+  uint64_t write_buffer_int64;
+  unsigned char write_buffer_char[8];
+} write_buffer;
 
-static int remainder_count;
+HAL_StatusTypeDef internal_flash_write_doubleword(uint32_t destination, uint64_t source)
+{
+  HAL_FLASH_Unlock();
+  if(HAL_FLASH_Program(FLASH_TYPEPROGRAM_DOUBLEWORD, destination, source) != HAL_OK)
+  {
+      HAL_FLASH_Lock();
+      return HAL_ERROR;
+  }
+  HAL_FLASH_Lock();
+  return HAL_OK;
+}
 
 /* Write firmware into internal flash.  */
 HAL_StatusTypeDef internal_flash_write(
```

Comment:
>internal_flash_write [](http://example.com/codeflow?start=18&length=20)

These changes will break the package download. Please consider use this function as is or create a new function for your application.

Reply:
I don't think these changes will break the package download... want to chat offline?

Reply:
Yes, let's chat about it.